### PR TITLE
Add CODEOWNERS file

### PR DESCRIPTION
**Pull request by nebril**

### CODE REVIEW

File: CODEOWNERS
Comment (ianvernon) [inline CODEOWNERS]: This is sort of an abstract concept, as it covers how endpoints are regenerated / when they are regenerated, which occurs in the `daemon` and `endpoint` directories.

File: CODEOWNERS
Comment (ianvernon) [inline CODEOWNERS]: I've informed @rlenglet of this offline on Slack - we need to wait for his approval for this as we volunteered his name :-)

File: CODEOWNERS
Comment (ianvernon) [inline CODEOWNERS]: I recommend @rlenglet and @manalibhutiyani.

**Review by ianvernon — COMMENTED** [review]
Some comments.

File: CODEOWNERS
Comment (rlenglet) [reply CODEOWNERS]: LGTM

File: CODEOWNERS
Comment (rlenglet) [inline CODEOWNERS]: LGTM

File: CODEOWNERS
Comment (rlenglet) [reply CODEOWNERS]: Yes, please add @rlenglet, @ashwinp, and @manalibhutiyani.

**Review by rlenglet — CHANGES_REQUESTED** [review]

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: I will create github groups to make this easier. What convention do we want to use for naming?

@grp-bpf? @bpf? @team-bpf?

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: Also add `examples/kubernetes/` and `examples/minikube/`

**Review by tgraf — CHANGES_REQUESTED** [review]

File: CODEOWNERS
Comment (manalibhutiyani) [reply CODEOWNERS]: My vote: @team-bpf

Comment (ianvernon) [reply CODEOWNERS]: +1, for prefixing with "team-"

Comment (tgraf) [inline CODEOWNERS]: @cilium/bpf

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/ci

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/kvstore

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/kubernetes

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/policy

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/docker

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/proxy

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/docs

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/vendor

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/agent

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/monitor

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/docs

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/cli

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/api

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/janitors

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: We will need to separate out this code first

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/contributing

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/endpoint

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: @cilium/loadbalancer

**Review by tgraf — CHANGES_REQUESTED** [review]

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: Let's use `bpf/` here and in all other places consistently. This ensures that subdirectories get matched as well. `bpf/*` will only match one nesting level.

**Review by tgraf — CHANGES_REQUESTED** [review]

File: CODEOWNERS
Comment (tgraf) [inline CODEOWNERS]: + pkg/proxy/accesslog/
+ monitor/payload/

This will ensure that  @cilium/api covers all code which external components depend on.

**Review by tgraf — CHANGES_REQUESTED** [review]

File: CODEOWNERS
Comment (borkmann) [inline CODEOWNERS]: Most parts are in `bpf/` for XDP and from golang side the main bits are in `pkg/maps/cidrmap/cidrmap.go`, `pkg/policy/prefilter.go`, `daemon/prefilter.go`.

File: CODEOWNERS
Comment (borkmann) [inline CODEOWNERS]: bpf/*.sh for the most part

**Comment by nebril** [comment]
Thanks for comments and suggestions. I fixed what I could, still have several questions:

@tgraf: should `Agent -> Kernel interaction` have a separate group? If yes, please create one consisting of @scanf and @borkmann. 

@jrajahalme please let me know if I missed any envoy-specific files/directories

@jrfastab is there any kproxy related code in cilium ATM?

@cilium/endpoint : where is endpoint generation and restore code?

File: CODEOWNERS
Comment (ianvernon) [inline CODEOWNERS:2]: I like this tag name :-)

File: CODEOWNERS
Comment (ianvernon) [inline CODEOWNERS]: `daemon/services.*` and `daemon/loadbalancer.*`.

File: CODEOWNERS
Comment (ianvernon) [inline CODEOWNERS]: This is difficult to create an 'owner' of because endpoint regeneration is triggered by the daemon based off of various events occurring. I suggest we get rid of this 'endpoint generation' owner and make it owned by those who manage `daemon/` and `endpoint/`. @tgraf what do you think?

File: CODEOWNERS
Comment (ianvernon) [inline CODEOWNERS]: See above comment for line 84.

**Review by ianvernon — CHANGES_REQUESTED** [review]

**Comment by tgraf** [comment]
> @tgraf: should Agent -> Kernel interaction have a separate group? If yes, please create one consisting of @scanf and @borkmann.

@cilium/bpf-loader

File: CODEOWNERS
Comment (tgraf) [reply CODEOWNERS]: I think we should isolate this code eventually. We can defer this dedicated owner until that has happened.

**Comment by nebril** [comment]
I think I fixed most issues, please re-review.

**Review by borkmann — APPROVED** [review]
lgtm, we should still add a @cilium/bpf-docs group, but can be done later as well.

**Review by ianvernon — APPROVED** [review]
LGTM!

**Review by rlenglet — APPROVED** [review]

File: CODEOWNERS
Comment (aanm) [inline CODEOWNERS:31]: examples/kubernetes-ingress/ @cilium/kubernetes

File: CODEOWNERS
Comment (aanm) [inline CODEOWNERS]: Do we have this directory?

**Review by aanm — CHANGES_REQUESTED** [review]

File: CODEOWNERS
Comment (nebril) [reply CODEOWNERS]: Good catch.

File: CODEOWNERS
Comment (aanm) [inline CODEOWNERS:81]: Also `daemon/endpoint.go`?

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by aanm — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Pull request merged**
